src/CommunityPatch/Patches/MenuWhenEncountingAnArmyPatch.cs

```diff
@@ -0,0 +1,105 @@
+  public sealed class MenuWhenEncounteringAnArmyPatch : PatchBase<MenuWhenEncounteringAnArmyPatch> {
+
+    public override bool Applied { get; protected set; }
+
+    private static readonly MethodInfo TargetMethodInfo =
+      typeof(PlayerEncounter)
+        .GetMethod("DoMeetingInternal", BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.DeclaredOnly);
+
+    private static readonly MethodInfo PatchMethodInfo = typeof(MenuWhenEncounteringAnArmyPatch)
+      .GetMethod(nameof(Prefix), BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.DeclaredOnly);
+
+    private static readonly byte[][] Hashes = {
+      new byte[] {
+        // e1.0.11
+        0x18, 0xC5, 0x16, 0xD2, 0x46, 0x07, 0xBB, 0x08,
+        0xAE, 0xC3, 0x6D, 0xE6, 0xFA, 0xCD, 0x2E, 0x63,
+        0x0E, 0x1A, 0x73, 0xB6, 0x76, 0xF9, 0xAD, 0x0E,
+        0x3F, 0xD9, 0xFA, 0x7A, 0x68, 0xFB, 0xEB, 0xE7
+      }
+    };
+
+    public override void Reset(){}
+
+    public override IEnumerable<MethodBase> GetMethodsChecked() {
+      yield return TargetMethodInfo;
+    }
+
+    public override void Apply(Game game) {
+      if (Applied) return;
+
+      CommunityPatchSubModule.Harmony.Patch(TargetMethodInfo,
+        prefix: new HarmonyMethod(PatchMethodInfo));
+      Applied = true;
+    }
+
+    public override bool? IsApplicable(Game game) {
+      var patchInfo = Harmony.GetPatchInfo(TargetMethodInfo);
+      if (AlreadyPatchedByOthers(patchInfo))
+        return false;
+
+      var hash = TargetMethodInfo.MakeCilSignatureSha256();
+      return hash.MatchesAnySha256(Hashes);
+    }
+    
+    private static readonly FieldInfo EncounteredPartyField = typeof(PlayerEncounter).GetField("_encounteredParty", BindingFlags.Instance | BindingFlags.NonPublic);
+
+    private static readonly FieldInfo MapEventStateField = typeof(PlayerEncounter).GetField("_mapEventState", BindingFlags.Instance | BindingFlags.NonPublic);
+    
+    private static readonly FieldInfo StateHandledField = typeof(PlayerEncounter).GetField("_stateHandled", BindingFlags.Instance | BindingFlags.NonPublic);
+    
+    private static readonly FieldInfo DefenderPartyField = typeof(PlayerEncounter).GetField("_defenderParty", BindingFlags.Instance | BindingFlags.NonPublic);
+    
+    private static readonly FieldInfo MeetingDoneField = typeof(PlayerEncounter).GetField("_meetingDone", BindingFlags.Instance | BindingFlags.NonPublic);
+    
+    
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static bool Prefix(PlayerEncounter __instance) {
+      if (EncounteredPartyField == null || MapEventStateField == null || StateHandledField == null || DefenderPartyField == null || MeetingDoneField == null) {
+        CommunityPatchSubModule.Error($"{typeof(MenuWhenEncounteringAnArmyPatch).Name}: Could not locate all of necessary private fields for patching." + Environment.NewLine);
+        return true;
+      }
+      
+      PartyBase party1 = (PartyBase) EncounteredPartyField.GetValue(__instance);
```

Comment:
party1 vs party2 aren't very descriptive and can make it a bit harder to follow imo.

Reply:
Tbh I left the vanilla implementation names.

Reply:
We can refactor the naming to attacker/defender or whatever this after it gets merged to dev, or...